text/####-error-mitigation.md

```diff
@@ -0,0 +1,235 @@
+This solution may be overly complex as the user has to calibrate himself the stretched gates.
+The calibration of stretched gates, could be automated to simplify the task for the user, but this would not decrease the run time.
+Qiskit-ignis would most likely need to be involved ontop of qiskit-aqua and qiskit-terra.
+
+**The advantages of this method are:**
+- It is very flexible.
+- This allows error mitigation to be applied on single-qubit and two-qubit gates.
+- The stretch factors may be chosen as a function of the depth of the quantum circuit.
+
+**The limitations of this method are**:
+- It requires a lot of knowledge from the user.
+- It requires that the user run many calibration jobs in addition to his quantum circuit.
+
+## Detailed Design
+Here we focus on the implementation details of the Backend constrained error mitigation.
+An overview of this method is shown in the figure below.
+
+![](202001_ErrorMitigation_summary.png)
+
+The backend will have a set of calibrated gates with different stretch factors that will be made available to Qiskit through the config file.
+A backend may implement as many stretch factors as is deemed reasonable by those who maintain the backend.
+A backend may also have the choice to not implement error mitigation at all.
+Ideally, each gate on the backend will be available with the same set of stretch factors.
+However, this may not always be the case.
+For example, a 20 qubit device may only have stretched gates calibrated for qubits zero to four.
+Therefore, the stretch factors will be specified in the `gateconfig` schema (see `qiskit/schemas/backend_configuration_schema.json`) which will be updated as follows
+```
+"gateconfig": {
+    "type": "object",
+    "required": ["name", "parameters", "qasm_def"],
+    "properties": {
+        ...
+        "stretch_factor": {
+            "type": "string"
+            "description": "The stretch factor of the gate for error mitigation (if supported)."
+        }
+    }
+}
+```
+To facilitate the user experience, helper functions that show the available stretch factors, will be created.
+First, we define that a given qubit has stretch factor `c` available if all gates that involve this qubit are available with this stretch factor.
+The user will thus be able to query the backend to find out which stretch factors are available on a given list of qubits:
+```
+def available_stretch_factors(self, qubits: List) -> List:
+    """Returns a list of stretch factors available for the given qubits."""
+```
+For example to find the stretch factors available to qubits 0, 1, and 3, the user would do `backend.available_stretch_factors([0, 1, 3])`.
+Similarly, the user may query the backend to find out which qubits support a given list of stretch factors.
+```
+def available_qubits(self, stretch_factors: List) -> List:
```

Comment:
To use this function one should know apriori what stretch factors there are. So I'm not sure it provides a lot of value over the methods the previous method.

Reply:
You are correct. I think `available_stretch_factors` will be more useful. I suggested adding `available_qubits` for completeness but we can also decide not to implement it and only implement it if we see a need for it later on.